### ISSUE TITLE
如何压缩pdf里面的图片 how to compress pdf images 

### ISSUE THREAD

**Issue by qq358678184** [J-F-Liu/lopdf#283]
如何压缩pdf里面的图片 how to compress pdf images

**Comment by Heinenen** [comment]
```rs
let mut img = xobject::image("assets/pdf_icon.jpg").unwrap();
img.compress().unwrap();
doc.insert_image(page_id, img, (100.0, 210.0), (400.0, 225.0)).unwrap();
```
This should losslessly compress the image, using the deflate algorithm.
If you want to decrease the PDF size further, you can look into the `image` crate to preprocess the images that you want to have in the PDF.

**Comment by qq358678184** [comment]
你好是这样的，pdf里面有许多图片，需要压缩里面的图片，这样pdf也就压缩了，而不是压缩图片再插入进去



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年11月17日(周日) 晚上8:00
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [J-F-Liu/lopdf] 如何压缩pdf里面的图片 how to compress pdf images  (Issue #283)




 
Closed #283 as completed.

**Comment by Heinenen** [comment]
So I assume my solution worked for you?
If not, feel free to reopen the issue.
